src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractDatasetCommand.java

```diff
@@ -212,4 +221,28 @@ public void setDataset(Dataset dataset) {
     protected Timestamp getTimestamp() {
         return timestamp;
     }
+
+    protected void checkSystemMetadataKeyIfNeeded(DatasetVersion newVersion, DatasetVersion persistedVersion) throws IllegalCommandException {
+        Set<MetadataBlock> changedMDBs = DatasetVersionDifference.getBlocksWithChanges(newVersion, persistedVersion);
+        changedMDBs.forEach(mdb -> {
+            logger.fine(mdb.getName() + " has been changed");
+        });
+        
+        String smdbString = JvmSettings.METADATA_BLOCK_SYSTEM_METADATA_KEYS.lookupOptional().orElse(null);
+        if (smdbString != null) {
+            JsonObject systemMetadataBlocks = JsonUtil.getJsonObject(smdbString);
```

Comment:
How about adding a converter instead of fiddling with the String itself? 

It's not very hard to do and makes code here much nicer to read and write. :wink: https://download.eclipse.org/microprofile/microprofile-config-3.0/microprofile-config-spec-3.0.html#_adding_custom_converters